test/test_producer.py

```diff
@@ -40,3 +41,39 @@ def partitions(topic):
         topic = b"test-topic"
         producer.send_messages(topic, b'hi')
         assert client.send_produce_request.called
+
+    def test_producer_async_queue_overfilled_batch_send(self):
+        queue_size = 2
+        producer = Producer(MagicMock(), batch_send=True, maxsize=queue_size)
+
+        topic = b'test-topic'
+        partition = 0
+        message = b'test-message'
+
+        with self.assertRaises(Full):
+            message_list = [message] * (queue_size + 1)
+            producer.send_messages(topic, partition, *message_list)
```

Comment:
that's a good idea.. message_list is currently empty right now, so we could put some garbage in there with "A", "B" and verify that we have the right content